sdk/src/main/scala/com/horizen/SidechainSettings.scala

```diff
@@ -92,10 +93,16 @@ case class EthServiceSettings(
 
 case class AccountMempoolSettings(maxNonceGap: Int = 16,
                                   maxAccountSlots: Int = 16,
-                                  maxMemPoolSlots: Int = 6144){
+                                  maxMemPoolSlots: Int = 6144,
+                                  maxNonExecMemPoolSlots: Int = 1024){
   require(maxNonceGap > 0, s"Maximum Nonce Gap not positive: $maxNonceGap")
   require(maxAccountSlots > 0, s"Maximum Account Slots not positive: $maxAccountSlots")
-  require(maxMemPoolSlots > 0, s"Maximum Memory Pool Slots not positive: $maxMemPoolSlots")
+  require(maxMemPoolSlots >= MempoolMap.MaxNumOfSlotsForTx, s"Maximum Memory Pool Slots number should be at least " +
+    s"${MempoolMap.MaxNumOfSlotsForTx} but it is $maxMemPoolSlots")
+  require(maxNonExecMemPoolSlots >= MempoolMap.MaxNumOfSlotsForTx, s"Maximum Non Executable Memory Sub Pool Slots number " +
+    s"should be at least ${MempoolMap.MaxNumOfSlotsForTx} but it is $maxNonExecMemPoolSlots")
+  require(maxNonExecMemPoolSlots < maxMemPoolSlots, s"Maximum Non Executable Memory Sub Pool Slots " +
```

Comment:
`<` or `<=` ?
do we allow max non exec slots to be equal to max slots? I think, yes

Reply:
If max non exec is less than max slots, there will be always room for an executable tx and a malicious user won't be able to use non-exec txs to evict exec ones. Let's say the max slots = max non exec  = 10. Let's suppose that in mempool there are 4 exec txs and 6 non exec txs. I could create enough non exec txs (the exact number depends on how old are the exec txs) so to evict the 4 exec ones. 
But it is a complicated matter, maybe it is better to discuss it.

Reply:
it seems like 1 slot will not make a big difference.
It is a responsibility of the Node holder to choose the values in a way to keep some space for executables only.